nx/lib/nx/defn/expr.ex

```diff
@@ -213,15 +218,31 @@ defmodule Nx.Defn.Expr do
   """
   def while(initial, context, arg, condition, body) do
     [flatten_initial, flatten_arg, flatten_body] = clauses = flatten_clauses([initial, arg, body])
-    args = [flatten_initial, flatten_arg, condition, flatten_body]
+
+    Composite.traverse(condition, fn
+      %T{vectorized_axes: [_ | _]} ->
+        raise ArgumentError,
+              "condition for while cannot be vectorized, got: #{inspect(condition)}"
+
+      _ ->
+        :ok
+    end)
```

Comment:
```suggestion
    if condition.vectorized_axes != [] do
      raise ArgumentError,
            "condition for while cannot be vectorized, got: #{inspect(condition)}"
    end
```

I beleive condition is always a single expression.